ceph.spec.in

```diff
@@ -1262,6 +1262,8 @@ Summary:        MIB for SNMP alerts
 BuildArch:      noarch
 %if 0%{?suse_version}
 Group:          System/Monitoring
+Requires:       snmp-mibs
+BuildRequires:  snmp-mibs
```

Comment:
why `BuildRequires`?

Reply:
I believe it's required so that /usr/share/snmp is owned by a package at _build_ time, when rpmlint runs. But now that you've asked, I have to do another build to double check that I'm not mistaken :-)

Reply:
Confirmed. Here's the story:

- `BuildRequires: snmp-mibs` makes sure /usr/share/snmp exists at build time, so rpmlint can succeed (there's a package that owns that directory)
- `Requires: snmp-mibs` makes sure /usr/share/snmp exists when the package is installed. Without this, you can still install ceph-mib, but nothing owns /usr/share/snmp, so that latter directory doesn't get removed if you uninstall ceph-mib. Whereas, if snmp-mibs is installed, it owns that directory, so uninstalling snmp-mibs gets rid of that directory.

Now that I've gone through all that though, I'm actually leaning towards just making ceph-mib own /usr/share/snmp again, i.e. we can do this:

```diff --git a/ceph.spec.in b/ceph.spec.in
index 1671dddf709..614d50c6dc5 100644
--- a/ceph.spec.in
+++ b/ceph.spec.in
@@ -2594,6 +2594,7 @@ exit 0
 %config %{_sysconfdir}/prometheus/ceph/ceph_default_alerts.yml
 
 %files mib
+%attr(0755,root,root) %dir %{_datadir}/snmp
 %{_datadir}/snmp/mibs
 
 %changelog
```

Rationale: why pull in a whole extra package (snmp-mibs, which includes a bunch of other non-ceph MIB files) just to make a parent directory exist?

What do you think?

Reply:
Thank you Tim for the thorough investigation! TIL. And this makes sense to me also. Let me add Justin as a reviewer also. Actually, his original change introducing this package had the same line.

Reply:
LGTM

Reply:
I should've probably said explicitly that I agree with the proposed change here to remove snmp-mips and just attr the snmp/ directory